comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala

```diff
@@ -70,28 +73,54 @@ object HandleMessages {
       _ <- ConnectionsCell[F].addConnAndReport(peer)
     } yield handledWithoutMessage
 
-  def handleProtocolHandshake[F[_]: Monad: TransportLayer: ConnectionsCell: RPConfAsk: Log: Metrics](
+  // TODO: investigate what's with unused parameter - protocolHandshake
+  def handleProtocolHandshake[F[_]: Monad: TransportLayer: Log: ConnectionsCell: RPConfAsk: Metrics](
       peer: PeerNode,
       protocolHandshake: ProtocolHandshake
-  ): F[CommunicationResponse] =
-    for {
-      conf     <- RPConfAsk[F].ask
-      response = ProtocolHelper.protocolHandshakeResponse(conf.local, conf.networkId)
-      resErr   <- TransportLayer[F].send(peer, response)
-      _ <- resErr.fold(
-            kp(().pure[F]),
-            kp(
-              Log[F].info(s"Responded to protocol handshake request from $peer") >>
+  ): F[CommunicationResponse] = {
+    def isLocalAddress(host: String): Boolean =
+      Try(InetAddress.getByName(host))
+        .map { a =>
+          a.isAnyLocalAddress || a.isLinkLocalAddress || a.isLoopbackAddress || a.isMulticastAddress || a.isSiteLocalAddress
```

Comment:
There are many check functions in `InetAddress` class. Why you decided this number of functions enough? Is it from some example from internet? If yes, please point me. Just interesting.

Reply:
This is copied from the removed code when this check was on the wrong place just to check local host.
https://github.com/rchain/rchain/pull/3734/files#diff-7646bc517fdec1ecae36d2db4d6e0bccb79a000b2bdff158e19844d704fa5673L27-L36

Tests shows that detection is correct, but it might review if all are needed or some check include others.